Syntaxes/Nim.YAML-tmLanguage

```diff
@@ -2,572 +2,985 @@
-    - begin: (?<!\$)(\$)(for|while|case|of|when|if|else|elif)( )
-      end: (\{|\n)
-      captures:
-        '1': {name: keyword.operator.nim}
-        '2': {name: keyword.control.nim}
-      endCaptures:
-        '1': {name: plain}
-      patterns:
-      - {include: source.nim}
-    - match: (?<!\$)(\$[a-zA-Z0-9_]+)
-      name: keyword.operator.nim
-    - {include: text.html.markdown.multimarkdown}
+    patterns:
+      - begin: '\{(?!\.)'
+        beginCaptures:
+          '0': {name: punctuation.section.pattern.begin.nim}
+        end: '}'
+        endCaptures:
+          '0': {name: punctuation.section.pattern.end.nim}
+        name: meta.pattern.nim
+        patterns:
+          - include: $self
 
-repository:
-  escaped_char:
-    patterns:
-    - name: constant.character.escape.newline.nim
-      match: \\[nN]
-    - name: constant.character.escape.carriagereturn.nim
-      match: \\[cC]|\\[rR]
-    - name: constant.character.escape.linefeed.nim
-      match: \\[lL]
-    - name: constant.character.escape.formfeed.nim
-      match: \\[fF]
-    - name: constant.character.escape.tabulator.nim
-      match: \\[tT]
-    - name: constant.character.escape.verticaltabulator.nim
-      match: \\[vV]
-    - name: constant.character.escape.double-quote.nim
-      match: \\\"
-    - name: constant.character.escape.single-quote.nim
-      match: \\'
-    - name: constant.character.escape.chardecimalvalue.nim
-      match: \\[0-9]+
-    - name: constant.character.escape.alert.nim
-      match: \\[aA]
-    - name: constant.character.escape.backspace.nim
-      match: \\[bB]
-    - name: constant.character.escape.escape.nim
-      match: \\[eE]
-    - name: constant.character.escape.hex.nim
-      match: \\[xX][0-9A-Fa-f]{2}
-    - name: constant.character.escape.backslash.nim
-      match: \\\\
-...
+  param-list:
+    patterns:
+      - begin: '\('
+        beginCaptures:
+          '0': {name: punctuation.section.parameters.begin.nim}
+        end: '\)'
+        endCaptures:
+          '0': {name: punctuation.section.parameters.end.nim}
+        name: meta.function.parameters
+        patterns:
+          - match: '[A-Za-z\x80-\xff](?:_?[\dA-Za-z\x80-\xff])*|_|`[^;,\n`]+`'
+            name: variable.parameter.nim
+          - match: '[,;]'
+            name: punctuation.separator.nim
+          - begin: '(:)|(=)'
+            beginCaptures:
+              '1': {name: punctuation.separator.annotation.nim}
+              '2': {name: keyword.operator.assignment.nim}
+            end: (?=[,;\)])
+            patterns:
+              - include: $self
+
+  invalid-names:
+    patterns:
+      - match: '(?x:addr|asm|bind|block|break|case|cast|concept|const|continue|converter
+        |defer|discard|distinct|do|elif|else|end|enum|except|export|finally|for
+        |from|func|if|import|include|interface|iterator|let|macro|method|mixin
+        |object|of|out|proc|ptr|raise|ref|return|static|template|try|tuple|type
+        |using|var|when|while|yield)\b'
```

Comment:
Because this group is marked by the `x` flag, I believe it can be reformatted:
```suggestion
      - match: |
          (?x:
            addr      |
            asm       |
            bind      |
            block     |
            break     |
            case      |
            cast      |
            concept   |
            const     |
            continue  |
            converter |
            defer     |
            discard   |
            distinct  |
            do        |
            elif      |
            else      |
            end       |
            enum      |
            except    |
            export    |
            finally   |
            for       |
            from      |
            func      |
            if        |
            import    |
            include   |
            interface |
            iterator  |
            let       |
            macro     |
            method    |
            mixin     |
            object    |
            of        |
            out       |
            proc      |
            ptr       |
            raise     |
            ref       |
            return    |
            static    |
            template  |
            try       |
            tuple     |
            type      |
            using     |
            var       |
            when      |
            while     |
            yield
          )\b
```

Reply:
But why ?